src/libraries/System.Runtime.InteropServices.JavaScript/tests/System.Runtime.InteropServices.JavaScript.UnitTests/System/Runtime/InteropServices/JavaScript/WebWorkerTest.cs

```diff
@@ -0,0 +1,209 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Threading.Tasks;
+using System.Threading;
+using System.Net.Http;
+using Xunit;
+using System.IO;
+using System.Collections.Generic;
+using System.Net.WebSockets;
+using System.Text;
+
+namespace System.Runtime.InteropServices.JavaScript.Tests
+{
+    // TODO test:
+    // JSWebWorker.RunAsync with CancellationToken
+    // JSExport 2x
+    // JSExport async
+    // timer
+    // GC + finalizer + dispose
+    // lock
+    // thread allocation, many threads
+    // TLS
+    // ProxyContext flow, child thread, child task
+    // use JSObject after JSWebWorker finished
+    // JSWebWorker JS setTimeout till after close
+    // WS on JSWebWorker
+    // Yield will hit event loop 3x
+    // HTTP continue on TP
+    // WS continue on TP
+    // event pipe
+    // FS
+
+    public class WebWorkerTest
+    {
+        #region executor threads
+
+        public static IEnumerable<object[]> GetTargetThreads()
+        {
+            yield return new object[] { new Executor(ExecutorType.Main) };
+            yield return new object[] { new Executor(ExecutorType.JSWebWorker) };
+            yield return new object[] { new Executor(ExecutorType.ThreadPool) };
+            yield return new object[] { new Executor(ExecutorType.NewThread) };
+            yield break;
+        }
+
+        public static IEnumerable<object[]> GetTargetThreads2x2()
+        {
+            yield return new object[] { new Executor(ExecutorType.Main), new Executor(ExecutorType.Main) };
```

Comment:
you could use a pair of loops for this, but it isn't super important to me (it would be a good way to avoid having a typo if someone changes this later though)